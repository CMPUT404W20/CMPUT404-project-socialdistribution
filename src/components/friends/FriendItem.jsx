import React, { Component } from "react";
import "../../styles/friends-notices-search/FriendItem.scss";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import Col from "react-bootstrap/Col";

class FriendItem extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }


  render() {
    const {
      username, userID, handleUnfollow, host,
    } = this.props;
    return (
      <Col md={6}>
        <div className="friend-item-wrapper">
          <Link
            to={{
              pathname: `/profile/${username}`,
              state: { username, userID, host },
            }}
            className="username-link"
          >
            {username}
          </Link>
          <DropdownButton
            id="friend-status"
            title="FRIENDS"
            drop="down"
            alignRight
          >
            <Dropdown.Item onClick={() => handleUnfollow(userID)}>Unfriend</Dropdown.Item>
          </DropdownButton>
        </div>
      </Col>
    );
  }
}

FriendItem.propTypes = {
  username: PropTypes.string.isRequired,
  host: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  handleUnfollow: PropTypes.func.isRequired,
};

export default FriendItem;
