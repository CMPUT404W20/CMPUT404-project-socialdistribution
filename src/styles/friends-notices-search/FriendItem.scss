@import "../common.scss";
.friend-item-wrapper {
    border: 1px solid $border-color;
    box-sizing: border-box;
    padding: 0 1rem 0 1rem;
    border-radius: 4px;
    height: 4rem;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 1rem;
    .username-link {
        text-decoration: none;
        font-weight: 600;
        font-size: 12pt;
        color: black;
    }
    .username-link:hover {
        color: $accent-color;
    }
    #friend-status {
        @extend %button;
        
        &:not(:disabled) {
            &:hover, &:focus {
                box-shadow: none;
            }
        }
    }
    .dropdown-menu {
        @extend %dropdown;
    }
}