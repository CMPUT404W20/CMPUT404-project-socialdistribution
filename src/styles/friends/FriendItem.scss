.friend-item-wrapper {
    border: 1px solid #BFBFBF;
    box-sizing: border-box;
    border-radius: 4px;
}