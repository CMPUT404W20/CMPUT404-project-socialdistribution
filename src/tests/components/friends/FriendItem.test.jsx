import React from "react";
import { shallow } from "enzyme";
import FriendItem from "../../../components/friends/FriendItem";

describe("FriendItem Component", () => {
  it("should render correctly", () => {
    const component = shallow(<FriendItem id="" username="" handleUnfollow={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
