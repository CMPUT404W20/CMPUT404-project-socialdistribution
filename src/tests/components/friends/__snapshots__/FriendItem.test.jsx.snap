// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendItem Component should render correctly 1`] = `
<Col
  md={6}
>
  <div
    className="friend-item-wrapper"
  >
    <Link
      className="username-link"
      to={
        Object {
          "pathname": "/profile/",
          "state": Object {
            "host": "",
            "userID": "",
            "username": "",
          },
        }
      }
    />
    <DropdownButton
      alignRight={true}
      drop="down"
      id="friend-status"
      title="FRIENDS"
    >
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        onClick={[Function]}
      >
        Unfriend
      </DropdownItem>
    </DropdownButton>
  </div>
</Col>
`;
